59-bottlenose-dolphin/technical_specification.md

```diff
@@ -0,0 +1,121 @@
+  - How should the directory structure of the repository look like?
+  - How and to what extend do we document the new codebase?
+  - Are there any standards or tools that we should use for the documentation (JSDoc, Storybook)?
+  - How do we enforce that e.g. all classes and methods are annotated with some minimal documentation?
+  - Can we use some kind of automated documentation?
+  - Shall we provide some high-level documentation that can be used to onboard new developers?
+
+- Authentication:
+  - How can OIDC, 2FA, and user state management be integrated?
+  - Should we use a client-side library [angular-auth-oidc-client](https://github.com/damienbod/angular-auth-oidc-client) or [oidc-client-ts](https://github.com/authts/oidc-client-ts) or [angular-oauth2-oidc](https://github.com/manfredsteyer/angular-oauth2-oidc)?
+  - Should we instead move the OIDC client completely to the backend (auth adapter) for improved security, as [recommended by Philippe De Ryck](https://www.youtube.com/live/mORR3hpMaJQ)?
+
+- Design system:
+  - Which design system (Bootstrap, Material, Ant, Tailwind, Clarity, Carbon, Fluent, Chakra, Foundation) do we want to use? See our earlier evaluation [here](https://github.com/ghga-de/adrs/pull/5/).
+  - While Angular Material looks like a natural choice, is it suitable for a mostly non-mobile application like ours with high information density? Particularly, do we need data grids (tables) and how well are they supported?
+  - Settle on a component library. See our earlier evaluation [here](https://github.com/ghga-de/adrs/pull/9). Did we miss anything, is there anything new?
+  - How important are responsive design and accessibility to us? How well is this supported by the chosen system/library?
+  - Do we want to use plain CSS or SASS or SCSS syntax?
+  - Do we want to design pages upfront, creating mockups or wireframes? If yes, which tools for the UI/UX design do we want to use?
+
+- State management and other tooling:
+  - Shall we use a dedicated state management solution (like NgRx, Akita, Elf or NgxZustand)? Or start with a NGRX signal store or simple custom services to manage state?
+  - Should we primarily use RxJS or Signals?
+  - Shall we make use of a caching library for HTTP requests like NgHttpCaching or cashew?
+  - Are there any client libraries or development tools that we should make use of?
+
+- Deployment and configuration:
+  - How can we pass a runtime configuration similarly as we do with our microservices?
+  - How can we inject the configuration into the application without rebuilding the complete application, as we currently do for the React app?
+  - How can we rewrite the build and run steps from Python to JavaScript so that we do not need to install Python in the Docker container anymore?
+  - What is the best way to serve the SPA in production? We currently use the "serve" package from npm. Is that still up to date and recommended?
```

Comment:
`ng serve`, not `npm serve` no?
Also nowadays Angular docs use `npm start`

Reply:
With "currently" I'm referring to the react application. No, `ng serve` or `npm start` is used to run the dev server. This question is asking about production which should probably use a different web server. Clarified in 39eb3647.